active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,277 @@
+provider 'br/public:az@1.2.3' // uses provider aliasing defined in bicepconfig.json
+```
+
+We also currently support the following syntax for "built-in" providers:
+
+```bicep
+provider 'sys@1.0.0'
+```
+
+In the above cases, the provider can be referenced by using its symbol which is derived from the segment to the left of the '@' sign. (`az` and `sys` for the declarations above)
+
+Both syntaxes also support aliasing of that symbol reference:
+
+```bicep
+provider 'br/public:az@0.2.3' as myAz
+provider 'sys@1.0.0' as mySys
+```
+
+The current syntax (shown above) presents several challenges:
+
+1. The new syntax is quite verbose, and cannot be controlled centrally. For users with a large set of files, picking up the latest version of `az` will require modifying every single file.
+1. The aliasing (assuming that the last segment of the registry namespace == the provider name) & versioning (having an `@` character instead of a `:` which is familiar to module users) behavior of the new syntax can be confusing.
+1. Its deceiving and confusing to allow `sys` to have a pinned versionis fixed version `1.0.0` - this **should** imply that there are no changes between Bicep versions, which isn't realistic.
+
+- It's an "either/or" - either users follow the current behavior (`az` imported by default) **or** they're using the new syntax. We don't have a plan for how to combine the two.
+- With [Radius](https://github.com/radius-project/radius), it's conceivable that we may want users to **not** have `az` imported by default.
+
+### Proposed Changes
+
+- [Resolution of namespace identifiers by VSCode extension when inspecting published module sources](#resolution-of-namespace-identifiers-by-vscode-extension-when-inspecting-published-module-sources)
+- [Wildcard semantics for provider versions in `bicepconfig.json`](#wildcard-semantics-for-provider-versions-in-bicepconfigjson)
+
+## Detailed design
+
+### Client side changes
+
+#### Add a new syntax for the provider import declaration statements: `provider {providerName} [as {optionalAlias}]`
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+provider az
+provider kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+
+  "providers": {
+    "az": {
+      // A package reference string (below) gestures Bicep to source the types from the registry address.
+      "source": {
+        "registry": "mcr.microsoft.com/bicep/providers",
+        "version": "0.2.3",
+      },
+      "isDefaultImport": true // Gestures Bicep to implicitly import the namespace identifier to the global scope
+    },
+    "kubernetes": {
+      // Using the value "builtin" (below) gestures Bicep to source types from the statically compiled NuGet package reference
+      "source": {
+        "builtin": true
```

Comment:
Should we change the casing to `builtIn` since this is equivalent to built-in in normal English.

Reply:
yeah, that works